src/main/java/net/mcreator/ui/dialogs/imageeditor/ResizeDialog.java

```diff
@@ -55,8 +57,17 @@ public ResizeDialog(MCreator window, Canvas canvas, Layer layer, VersionManager
 		cancel.addActionListener(e -> setVisible(false));
 
 		ok.addActionListener(e -> {
+			CanvasResize canvasResize = null;
 			layer.resize((int) width.getValue(), (int) height.getValue(), type.isSelected());
-			versionManager.addRevision(new Modification(canvas, layer));
+			if (affectCanvas.isSelected() && (canvas.getWidth() < layer.getWidth()
+					|| canvas.getHeight() < layer.getHeight())) {
+				canvasResize = canvas.setSize(Math.max(canvas.getWidth(), layer.getWidth()),
+						Math.max(canvas.getHeight(), layer.getHeight()));
+			}
+			if (canvasResize != null)
+				versionManager.addRevision(new Modification(canvasResize, layer));

```

Comment:
Check the main comment. This is also where most of the changes need to be made (generating a fresh UUID and assigning it to both changes)

Reply:
:heavy_check_mark: Fixed